smack-websocket/src/main/java/org/jivesoftware/smack/websocket/XmppWebsocketTransportModuleDescriptor.java

```diff
@@ -0,0 +1,74 @@
+/**
+ *
+ * Copyright 2020 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smack.websocket;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.jivesoftware.smack.ConnectionConfiguration.SecurityMode;
+import org.jivesoftware.smack.c2s.ModularXmppClientToServerConnectionModule;
+import org.jivesoftware.smack.c2s.ModularXmppClientToServerConnectionModuleDescriptor;
+import org.jivesoftware.smack.c2s.internal.ModularXmppClientToServerConnectionInternal;
+import org.jivesoftware.smack.fsm.StateDescriptor;
+import org.jivesoftware.smack.websocket.XmppWebsocketTransportModule.EstablishingWebsocketConnectionStateDescriptor;
+
+public final class XmppWebsocketTransportModuleDescriptor extends ModularXmppClientToServerConnectionModuleDescriptor {
+
+    private boolean usingHttps = false;
+
+    public XmppWebsocketTransportModuleDescriptor(Builder builder) {
+        this.usingHttps = builder.usingHttps;
+    }
+
+    public boolean isUsingHttps() {
+        return usingHttps;
+    }
+
+    @Override
+    protected Set<Class<? extends StateDescriptor>> getStateDescriptors() {
+        Set<Class<? extends StateDescriptor>> res = new HashSet<>();
+        res.add(EstablishingWebsocketConnectionStateDescriptor.class);
+        return res;
+    }
+
+    @Override
+    protected ModularXmppClientToServerConnectionModule<? extends ModularXmppClientToServerConnectionModuleDescriptor> constructXmppConnectionModule(
+            ModularXmppClientToServerConnectionInternal connectionInternal) {
+        return new XmppWebsocketTransportModule(this, connectionInternal);
+    }
+
+    public static final class Builder extends ModularXmppClientToServerConnectionModuleDescriptor.Builder {
+
+        boolean usingHttps = false;
+
+        protected Builder(
+                org.jivesoftware.smack.c2s.ModularXmppClientToServerConnectionConfiguration.Builder connectionConfigurationBuilder) {
+            super(connectionConfigurationBuilder);
+            connectionConfigurationBuilder.setSecurityMode(SecurityMode.disabled);
```

Comment:
Is there a reason for this? This scares me :P

Reply:
That line needs to be removed